inject/src/main/java/io/avaje/inject/spi/DBeanScope.java

```diff
@@ -175,10 +182,11 @@ private <T> List<T> listOf(Type type) {
   static <T> List<T> combine(List<T> values, List<T> parentValues) {
     if (values.isEmpty()) {
       return parentValues;
-    } else if (parentValues.isEmpty()) {
-      return values;
     } else {
-      values.addAll(parentValues);
+      if (parentValues.isEmpty()) {
```

Comment:
This (A) creates an empty if block which I really don't like and (B) isn't related to JSpecify so really shouldn't be in this PR?